src/opengl/platform/rendercontext_ogl_win.cpp

```diff
@@ -0,0 +1,113 @@
+#if defined(_WIN32)
+
+#include "rendercontext_ogl_win.h"
+#include "common/config.h"
+
+#include <GL/glew.h>
+#include <GL/wglew.h>
+#include <iostream>
+
+namespace {
+    PIXELFORMATDESCRIPTOR m_pixelFormat;
+    HGLRC m_renderContext;
+    HDC m_deviceContext;
+    HWND m_hwnd;
+}
+
+namespace dw {
+
+// TODO: Clean this up, the class looks like a mess
+RenderContextWin::RenderContextWin(const PlatformData& platformData) {
+    int errorCode;
+    HWND hwndFake = CreateWindowA("STATIC", "", WS_POPUP | WS_DISABLED, -32000, -32000, 0,
+                                  0, NULL, NULL, GetModuleHandle(NULL), 0);
+
+    // Initialize extensions
+    // Get the device context
+    m_deviceContext = GetDC(hwndFake);
+    // Set a temporary pixel format
+    errorCode = SetPixelFormat(m_deviceContext, 1, &m_pixelFormat);
+    // Create temporary render context
+    m_renderContext = wglCreateContext(m_deviceContext);
+    // Set the temporary rendering context as the current rnedering context for this
+    // window
+    errorCode = wglMakeCurrent(m_deviceContext, m_renderContext);
+    // Initialize the OpenGL extensions, we need a temp context to do so
+    GLenum err = glewInit();
+    if (err != GLEW_OK) {
+        std::cerr << "Direwolf: Failed to initialize extensions!";
+    }
+
+    // Release the temporary render context now that the extensions have loaded
+    wglMakeCurrent(nullptr, nullptr);
+    wglDeleteContext(m_renderContext);
+
+    // Release device context
+    ReleaseDC(hwndFake, m_deviceContext);
+
+    // Get the device context again
+    m_hwnd = static_cast<HWND>(platformData.windowHandle);
```

Comment:
Is this really safe? Casting a `void*`? Well since we control it I guess it is.
However, are there use cases on windows where the window handle is used and it doesn't have to be cast? If not I think it'd make more sense that the struct holds the HWND directly.
Isn't it advised to stay away from casts?

Reply:
it's safe as long as we don't give it something else than the HWND :D 
The thing is it's convenient to use a single struct for platformParams.. and cast platform specific stuff. I've seen this in other libraries. The alternative is to make platform params an interface and make platform specific structs that implement that interface.. maybe overkill for now but idk